explainer.md

```diff
@@ -215,6 +244,175 @@ third-party then signs either the exchanges in a package or the package as a
+
+### Signature verification
+
+Once the certificates are validated and enough of the
+`application/signed-exchange` resource is received to parse the claimed
+signed-exchange headers, the client extracts the logical URL from those headers
+and then tries to find a valid signature over the headers that is trusted for
+the claimed origin. If none of the signatures are valid, it either
+
+1. redirects to the logical URL as if the whole signed exchange were a 302
+   response, or
+2. fails with a network error.
+
+We're not yet certain which behavior is best. The first is slightly more
+resilient to clock skew, while the second encourages intermediates to get their
+implementations right.
+
+If the client does find a valid signature, it "releases" the headers for further
+processing and starts validating
+[mi-sha256](https://tools.ietf.org/html/draft-thomson-http-mice-02#section-2)
+records into a response stream as they arrive.
+
+### Prefetching stops here
+
+At this point, the client's behavior depends on whether the signed exchange was
+requested as a [prefetch](https://w3c.github.io/resource-hints/#prefetch). To
+satisfy the [privacy-preserving prefetch](#privacy-preserving-prefetch) use
+case, prefetches can't fully load the logical URL, which would create an HTTP
+cache entry that would be visible to the logical URL's server.
+
+Prefetches can and should process any `Link: <>; rel=preload` headers they find,
+as prefetches. If those point at signed exchanges, this process repeats.
+
+### No nested signed exchanges
+
+To limit the complexity of the implementation, we're currently planning to
+disallow signed exchanges that contain either signed exchanges or redirects.
+This may change if use cases come up or if the implementation turns out to be
+simpler than expected.
+
+### Caching the signed response
+
+If the signed exchange was requested as a navigation or subresource (i.e. *not*
+prefetches), the client tries to cache the request→response pair it contains.
+
+This *doesn't* happen if:
+
+* The signed exchange's request headers aren't sufficiently similar (TBD) to the
+  request headers the client would use for a normal request in the same context.
+  This prevents a malicious intermediate from sticking the wrong
+  content-negotiated resource in the HTTP cache.
+* There's a response in the HTTP (or any?) cache with a newer Date header
+  than the signed exchange's response. This prevents some downgrade attacks.
+
+In either of these cases, the client just skips to the redirect in the next
+step.
+
+If we're still here, the signed exchange is put into a "use-once" cache similar
+to the [preload cache](https://github.com/whatwg/fetch/issues/590) and, if the
+response headers allow it (and review of this explainer indicates it's a good
+idea), the [HTTP cache](https://tools.ietf.org/html/rfc7234).
+
+If we put the signed exchange in the HTTP cache, its freshness has to be bounded
```

Comment:
HTTP freshness can't be modified like this; any upstream caches (e.g., CDN) won't be aware of these special semantics. 

I think the two approaches you have are:
1. Don't do that -- i.e., make sure HTTP cache lifetime is always shorter than the signature expiration.
2. If a client encounters a fresh cached response with an expired signature, it can force-refresh using things like `Cache-Control: max-age` in requests. Be aware, though, that request cache directives are ignored by many...

Reply:
If I rewrite this sentence as "If we put the *content* of the signed exchange in the HTTP cache", does that fix things for you? I definitely don't mean that the envelope needs to have different cache behavior, just the signed bits we pull out of the envelope. (I'm looking for clearer words about this; distinguishing the envelope format from the logical HTTP exchange inside it has been tricky.)

I *think* that's ok because the upstream caches like CDNs can only put the contents of a signed exchange into their cache if they understand signed exchanges, which means they're modifying code anyway and can also modify it in this way. I could even be wrong about that because it'd be code owned by two different groups?